en/guides/codablerouting.md

```diff
@@ -15,172 +15,263 @@ redirect_from:
 
 <div class="titleBlock">
   <h1>Codable Routing</h1>
-  <p>How to use Codable Routing in Kitura</p>
 </div>
 
 
 
 ## Introduction
-Kitura 2.0 introduces “Codable Routing”. Here the router handlers are much like normal functions you might define elsewhere in your code: they take struct or class types as parameters, and respond with stuct or class types via a completion handler, with just an additional requirement that those types implement [Codable](https://developer.apple.com/documentation/swift/codable) from Swift 4 (hence the name).
+Kitura has two types of routing. Raw Routing, which works directly with Router Requests and Router Responses, and Codable Routing which works directly with Swift types, provided they conform to [Codable](https://developer.apple.com/documentation/swift/codable). 
 
-Codable Routing isn’t suitable for every use case and scenario, so Raw Routing is still available where you need the power and flexibility, but it is perfect for building REST APIs such as you might want to do to build a Backend For Frontend (BFF) for an iOS app.
+Codable Routing was made available in Kitura 2.0, before this release only Raw Routing was available. Kitura 2.0 supports both types of Routing, depending on what your application needs. You can implement both types of routing in your application as well. This offers a lot of flexibility on how you build your server. 
 
-## Building a Create API for POST requests
+For many, Codable Routing feels more 'natural' as you work directly with Swift types.  
+This guide shows you step by step how to set up Codable routes on a Kitura server. 
 
-Building a Codable Route for a POST API to implement storing a ToDo item requires only 4 lines of code. The first step is to register a handler for `POST` requests on “/todos”:
+## Let's build a Server!
 
-```swift
-router.post("/todos", handler: storeHandler)
-```
+### Getting started
+First we need a Kitura server to add our routes to. Kitura has a [command line interface](https://github.com/IBM-Swift/kitura-cli) which provides us with utilities for quickly creating a Kitura server.  
+> ![tip] You can find installation instructions for the command line interface in the [Getting Started](/{{ page.lang }}/starter/gettingstarted.html) section. 
 
-and then provide an implementation of the storeHandler:
+Once we've installed the Kitura command line interface we can open a terminal window and run the following:  
```

Comment:
Once you've installed the Kitura command-line interface, open a terminal window: